src/building.md

```diff
@@ -239,3 +239,76 @@ git clone https://github.com/spack/spack.git
 cd spack
 ./bin/spack install -v mfem
 ```
+
+## Building MFEM with CMake
+To build a serial form of MFEM with CMake first create a build directory. For example, using a build directory named `build`:
+```sh 
+mkdir build 
+cd build 
+```
+
+Run the CMake configuration on the MFEM source directory.
+```sh 
+cmake ..
+```
+
+Run the build command associated with the CMake configuration, specifying the number of parallel build tasks with the `-j` flag (4 tasks in this case). 
+```sh 
+cmake --build . -j 4
+```
+### Parallel build using CMake
+To build a parallel form of MFEM with CMake first build METIS and Hypre as described above.
+From the MFEM source directory, create a build directory. For example, using a build directory named `build`:
+```sh 
+cd mfem-4.5
+mkdir build 
+cd build 
+```
+
+Run the CMake configuration on the MFEM source directory using the `MFEM_USE_MPI` CMake variable to enable MPI. 
+This will automatically search for the system MPI implementation, the METIS installation (in `<mfem-source-dir>/../metis-4.0`), and Hypre installation (in `<mfem-source-dir/../hypre`).
+```sh 
+cmake .. -DMFEM_USE_MPI=YES
+```
+Run the build command associated with the cmake configuration, specifying the number of parallel build tasks with the `-j` flag (4 tasks in this case). 
+```sh 
+cmake --build . -j 4
+```
+
+### Alternate configuration steps
+To build with CUDA:
+```sh 
+cmake .. -DMFEM_USE_CUDA=YES
+```
+Note that this requires CMake 3.8 or newer
+
```

Comment:
````suggestion
To specify what CUDA architecture to target:
```sh
cmake .. -DCUDA_ARCH="sm_70"
```
The CUDA architecture is formatted as `sm_{CC}`, where CC is the GPU compute capability of the target GPU without the decimal point. A list of Nvidia GPU compute capabilities can be found in [the Nividia developers documentation](https://developer.nvidia.com/cuda-gpus).
````

TODO: currently I think MFEM's CMake setup doesn't allow targeting multiple CUDA architectures at once. I think we should fix this?

Reply:
There is a CMake 3.18 target property `CUDA_ARCHITECTURES` that allows semicolon separated lists https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html 

Right now it looks like mfem manually sets the `-arch` flag. From the [nvcc compiler documentation](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#gpu-architecture-arch) it seems like it might take comma separated lists if the format is the same as the `--gpu-code` flag. I don't have access to cuda enabled machines to try this out right now. 

The `all`, `all-major`, and `native` options for the architecture may be worth mentioning.

Reply:
I expanded support for `all`, `all-major`, and `native` as well as multiple specific cuda architectures in this PR: [pr4561](https://github.com/mfem/mfem/pull/4561).

Supported formats now include:
`-DCUDA_ARCH="all"`
`-DCUDA_ARCH="all-major"`
`-DCUDA_ARCH="native"`
`-DCUDA_ARCH="{ARCH1},{ARCH2},..."`
`-DCUDA_ARCH="{ARCH1};{ARCH2};..."`
where ARCHN can be either just the CC number (`70`, `86`, etc.), or optionally prefixed with `sm_` (`sm_70`, `sm_86`, etc.).

This should work even for CMake older than 3.18 (the current CMakeLists minimum version is 3.8).

Reply:
looks great, added some documentation for this!